### PR TITLE
add checks for valid PoWE statuses and add descriptive messages for E…

### PR DESCRIPTION
**Pull request by emina-str**
…xception

### CODE REVIEW

File: application/src/main/java/ch/admin/seco/alv/service/onlineform/application/npa/ProofOfWorkEffortsApplicationService.java [hunk 1]
```diff
@@ -98,6 +98,8 @@ public void handleNPABusinessCaseCancelledEvent(NPABusinessCaseCancelledEvent np
 			ProofOfWorkEfforts poWE = existingPoWE.get();
 			if (poWE.getWorkEfforts().isEmpty()) {
 				this.delete(poWE);
+			} else {
+				throw new IllegalStateException("ProofOfWorkEfforts with ID : " + poWE.getId() + " has WorkEfforts added to it. Can't be deleted.");
```
Comment (pavax) [inline application/src/main/java/ch/admin/seco/alv/service/onlineform/application/npa/ProofOfWorkEffortsApplicationService.java]: this check does not belong to the ApplicationService but should be applied in the delete() method

File: domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/npa/ProofOfWorkEffortsRepository.java [hunk 2]
```diff
@@ -44,7 +44,7 @@
 	@Query("SELECT p FROM ProofOfWorkEfforts p WHERE p.ownerUserId = :userId AND p.controlPeriod.value < :yearMonth ORDER BY p.controlPeriod.value DESC")
 	List<ProofOfWorkEfforts> findAllBefore(UserId userId, YearMonth yearMonth, Pageable pageable);
 
-	@Query("SELECT p FROM ProofOfWorkEfforts p WHERE p.nextSubmissionDate < :currentDate order by p.personNumber, p.controlPeriod.value")
+	@Query("SELECT p FROM ProofOfWorkEfforts p WHERE p.nextSubmissionDate < :currentDate AND p.status IN ('OPEN','RE_OPENED') order by p.personNumber, p.controlPeriod.value")
```
Comment (pavax) [inline domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/npa/ProofOfWorkEffortsRepository.java]: why did you add the status to the query?

Comment (emina-str) [reply domain/src/main/java/ch/admin/seco/alv/service/onlineform/domain/npa/ProofOfWorkEffortsRepository.java]: removed, we check the valid status later in submit method, so no need for "double check".

File: application/src/main/java/ch/admin/seco/alv/service/onlineform/application/npa/ProofOfWorkEffortsApplicationService.java (hunk 1 above)
Comment (emina-str) [reply application/src/main/java/ch/admin/seco/alv/service/onlineform/application/npa/ProofOfWorkEffortsApplicationService.java]: fixed.

File: application/src/main/java/ch/admin/seco/alv/service/onlineform/application/npa/ProofOfWorkEffortsApplicationService.java [hunk 3]
```diff
@@ -112,6 +109,9 @@ public void delete(ProofOfWorkEffortsId id) {
 	}
 
 	private void delete(ProofOfWorkEfforts proofOfWorkEfforts) {
+		if (!proofOfWorkEfforts.getWorkEfforts().isEmpty()) {
+			throw new IllegalStateException("ProofOfWorkEfforts with ID : " + proofOfWorkEfforts.getId() + " has WorkEfforts added to it. Can't be deleted.");
```
Comment (pavax) [inline application/src/main/java/ch/admin/seco/alv/service/onlineform/application/npa/ProofOfWorkEffortsApplicationService.java]: sometimes we use BusinessExceptions and sometimes IllegalStateException -> is there a rule?

Comment (emina-str) [reply application/src/main/java/ch/admin/seco/alv/service/onlineform/application/npa/ProofOfWorkEffortsApplicationService.java]: hm, I think we don't have any rule..

Comment (emina-str) [reply application/src/main/java/ch/admin/seco/alv/service/onlineform/application/npa/ProofOfWorkEffortsApplicationService.java]: Changed the Exception type for the ProofOfWorkEfforts specific problems, to a Business Exception type.
